pkg/api/v1alpha1/nutanixdatacenterconfig_types.go

```diff
@@ -43,6 +43,31 @@ type NutanixDatacenterConfigSpec struct {
 	// for the Nutanix Prism Central. The namespace for the secret is assumed to be a constant i.e. eksa-system.
 	// +optional
 	CredentialRef *Ref `json:"credentialRef,omitempty"`
+
+	// FailureDomains is the optional list of failure domains for the Nutanix Datacenter.
+	// +optional
+	FailureDomains []NutanixDatacenterFailureDomain `json:"failureDomains,omitempty"`
+}
+
+// NutanixDatacenterFailureDomain defines the failure domain for the Nutanix Datacenter.
+type NutanixDatacenterFailureDomain struct {
+	// Name is the unique name of the failure domain.
+	// Name must be between 1 and 64 characters long.
+	// It must consist of only lower case alphanumeric characters and hyphens (-).
+	// It must start and end with an alphanumeric character.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	Name string `json:"name"`
+
+	// Cluster is the Prism Element cluster name or uuid that is connected to the Prism Central.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster,omitempty"`
+
+	// Subnets holds the list of subnets identifiers cluster's network subnets.
+	// +kubebuilder:validation:Required
+	Subnets []NutanixResourceIdentifier `json:"subnets,omitempty"`
```

Comment:
can you add validatins for these similar to this https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/nutanixmachineconfig.go#L220-L232

Reply:
Fixed